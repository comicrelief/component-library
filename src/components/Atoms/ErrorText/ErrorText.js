import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Text from '../Text/Text';
import spacing from '../../../theme/shared/spacing';
import alertIcon from '../Input/assets/error-alert-icon.png';

const ErrorIcon = styled.span`
  display: inline-block;
  padding: ${spacing('m')} ${spacing('md')} ${spacing('md')};
  width: 100%;
  z-index: 1;
  margin-top: -10px;
  background-color: ${({ theme }) => theme.color('error_pink')};
  border-radius: ${spacing('sm')};
  font-weight: bold;

  :before {
    position: relative;
    content: '';
    top: 2px;
    margin-right: 6px;
    background: url(${alertIcon}) left 0/100% no-repeat;
    width: 18px;
    height: 19px;
    display: inline-block;
    vertical-align: top;
    color: ${({ theme }) => theme.color('red')};
  }

  @media ${({ theme }) => theme.breakpoint('small')} {
    max-width: 290px;
  }

`;

const ErrorText = ({ children, ...rest }) => (
  <Text {...rest} color="red">
    <ErrorIcon>{children}</ErrorIcon>
  </Text>
);

ErrorText.propTypes = {
  children: PropTypes.node.isRequired
};

export default ErrorText;
