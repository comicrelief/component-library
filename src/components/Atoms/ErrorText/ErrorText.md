# ErrorText

```jsx
<ErrorText size="error">This is an error</ErrorText>
```
