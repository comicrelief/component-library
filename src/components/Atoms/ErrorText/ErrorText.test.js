import React from 'react';
import 'jest-styled-components';

import renderWithTheme from '../../../../tests/hoc/shallowWithTheme';
import ErrorText from './ErrorText';

it('renders correctly', () => {
  const renderer = renderWithTheme(
    <ErrorText size="sm">This is an error</ErrorText>
  );
  expect(renderer.toJSON()).toMatchSnapshot();
});
