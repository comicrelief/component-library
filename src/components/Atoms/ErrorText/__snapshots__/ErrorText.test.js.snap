// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  color: #E52630;
  font-size: 0.875rem;
  line-height: 0.875rem;
  text-transform: inherit;
  line-height: normal;
  font-family: 'Montserrat',Helvetica,Arial,sans-serif;
}

.c1 {
  display: inline-block;
  width: 100%;
  z-index: 1;
  font-weight: bold;
  padding: 0.6rem 0 1rem;
}

<span
  className="c0"
  color="red"
  size="error"
>
  <span
    className="c1"
  >
    This is an error
  </span>
</span>
`;
